penguin/web/sse_events.py

```
@@ -284,12 +318,30 @@ def _remember_replay_event(core: Any, event: dict[str, Any]) -> None:
         del buffer[: len(buffer) - _MAX_REPLAY_EVENTS]
 
 
-def _replay_events_after(core: Any, last_event_id: str) -> list[dict[str, Any]]:
+def _replay_events_after(
+    core: Any,
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `core`

(ANN401)